src/js/components/navigation.js

```diff
@@ -129,6 +137,18 @@ navigation = behavior(
         }
       },
     },
+    keydown: {
+      [BODY]: keymap({ Escape: hideActiveNavDropdown }),
+    },
+    focusout: {
+      [NAV_PRIMARY](event) {
+        const nav = event.target.closest(NAV_PRIMARY);
+
+        if (!nav.contains(event.relatedTarget)) {
+          hideActiveNavDropdown();
```

Comment:
Created a new selector in `NAV_PRIMARY` because `NAV` can contain other elements that **aren't** part of the primary navigation menu.